Gems/Atom/RPI/Code/Source/RPI.Public/Pass/Pass.cpp

```diff
@@ -161,6 +185,9 @@ namespace AZ
                 m_path = ConcatPassName(m_parent->m_path, m_name);
                 m_flags.m_partOfHierarchy = m_parent->m_flags.m_partOfHierarchy;
 
+                m_parentDeviceIndex = m_parent->RecursiveGetDeviceIndex();
+                m_flags.m_parentDeviceIndexCached = true;
```

Comment:
It should not be necessary to call this recursively, since OnHierarchyChanged is already called recursively (the parent calls all children, so it goes the other way, which ensures that the parents are correctly updated, propagating the update down the hierarchy).